### PR TITLE
[Refactor] Use global constants for Ray versions and Ray image versions in go tests

### LINKED ISSUE #2618: [Refactor] Use global constants for Ray versions and Ray image versions in go tests
### Search before asking

- [X] I had searched in the [issues](https://github.com/ray-project/kuberay/issues) and found no similar feature requirement.

### Description

**Description:**
Several Ray version fields and Ray image versions in Go tests are hardcoded and should be replaced with global constants.

**Examples of Hardcoded Ray Versions:**
https://github.com/ray-project/kuberay/blob/aeba37e58813e0e415cb937fb0c977f2cfb981c4/ray-operator/controllers/ray/rayjob_controller_test.go#L53
https://github.com/ray-project/kuberay/blob/aeba37e58813e0e415cb937fb0c977f2cfb981c4/ray-operator/controllers/ray/rayservice_controller_test.go#L84

**Examples of Hardcoded Ray Image Versions:**
https://github.com/ray-project/kuberay/blob/aeba37e58813e0e415cb937fb0c977f2cfb981c4/ray-operator/controllers/ray/rayservice_controller_test.go#L107
https://github.com/ray-project/kuberay/blob/aeba37e58813e0e415cb937fb0c977f2cfb981c4/ray-operator/controllers/ray/raycluster_controller_unit_test.go#L228

**Proposed Solution:**
Replace these hardcoded values with the global constants defined in:
https://github.com/ray-project/kuberay/blob/aeba37e58813e0e415cb937fb0c977f2cfb981c4/ray-operator/test/support/defaults.go#L4-L5

This will enhance code maintainability across tests.

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by simotw**
## Why are these changes needed?

**Description:**
Several Ray version fields and Ray image versions in Go tests are hardcoded and should be replaced with global constants.

**Examples of Hardcoded Ray Versions:**
https://github.com/ray-project/kuberay/blob/aeba37e58813e0e415cb937fb0c977f2cfb981c4/ray-operator/controllers/ray/rayjob_controller_test.go#L53
https://github.com/ray-project/kuberay/blob/aeba37e58813e0e415cb937fb0c977f2cfb981c4/ray-operator/controllers/ray/rayservice_controller_test.go#L84

**Examples of Hardcoded Ray Image Versions:**
https://github.com/ray-project/kuberay/blob/aeba37e58813e0e415cb937fb0c977f2cfb981c4/ray-operator/controllers/ray/rayservice_controller_test.go#L107
https://github.com/ray-project/kuberay/blob/aeba37e58813e0e415cb937fb0c977f2cfb981c4/ray-operator/controllers/ray/raycluster_controller_unit_test.go#L228

**Proposed Solution:**
Replace these hardcoded values with the global constants defined in:
https://github.com/ray-project/kuberay/blob/aeba37e58813e0e415cb937fb0c977f2cfb981c4/ray-operator/test/support/defaults.go#L4-L5

This will enhance code maintainability across tests.

**Technical Decisions**

**Issue**  
Introducing `support.GetRayImage` to API test files, such as:
https://github.com/ray-project/kuberay/blob/aeba37e58813e0e415cb937fb0c977f2cfb981c4/ray-operator/apis/ray/v1/rayjob_types_test.go#L48
, causes an import cycle.

**Options**  
1. Create a separate package to resolve the import cycle.  
2. Avoid introducing `support.GetRayImage` in API test files.

**Decision**  
Option 2: Avoid introducing `support.GetRayImage` in API test files.  
The added complexity of creating a separate package is not justified for API test files.

## Related issue number

Closes #2618
## Checks

- [x] I've made sure the tests are passing.
- Testing Strategy
   - [x] Unit tests
   - [x] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: ray-operator/test/support/environment.go [hunk 1]
```diff
@@ -20,7 +22,11 @@ func GetRayVersion() string {
 }
 
 func GetRayImage() string {
-	return lookupEnvOrDefault(KuberayTestRayImage, RayImage)
+	rayImage := lookupEnvOrDefault(KuberayTestRayImage, RayImage)
+	if runtime.GOARCH == "arm64" && !strings.HasSuffix(rayImage, "-aarch64") {
+		rayImage = rayImage + "-aarch64"
```
Comment (kevin85421) [inline ray-operator/test/support/environment.go:27]: Add a log when explicitly modifying the user's image

**Comment by kevin85421** [comment]
@MortalHappiness can you review this PR? Thanks!

**Review by MortalHappiness — APPROVED** [review]
LGTM

File: ray-operator/test/support/environment.go (hunk 1 above)
Comment (simotw) [reply ray-operator/test/support/environment.go]: I have removed the arm chip support part after reconsideration. Thank you for the review!

**Review by kevin85421 — APPROVED** [review]

**Pull request merged**